### ISSUE TITLE
Add chunking logic in read method

### ISSUE THREAD

**Issue by sunank200** [astronomer/apache-airflow-provider-transfers#56]
**Describe the bug**
I tried an 11 GB file on a worker of 500 Mb and it got killed because of memory:
```
[2023-04-05, 21:03:34 UTC] {local_task_job.py:212} INFO - Task exited with return code Negsignal.SIGKILL
```

**Expected behavior**
The read method should only load chunks into memory. Currently, if there are multiple files in a folder each file is loaded into memory. But for scenarios when a single file is very large, we should have a logic to load only chunks at once.

**Comment by kaxil** [comment]
>Currently, if there are multiple files in a folder each file is loaded into memory

Yeah, entire file shouldn't be loaded in the memory. It can be one of the options but not the only option.

Flow (from fastest path to slowest):

1. Native path
2. Stream lines/bytes from source to destination via the worker
3. "naive" path - where we download all files from source to worker and then upload from worker to destination
